src/main/java/uk/gov/hmcts/cmc/claimstore/services/InterestRateService.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.cmc.claimstore.services;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.domain.exceptions.BadRequestException;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+
+@Component
+public class InterestRateService {
+
+    public BigDecimal calculateRate(LocalDate fromDate, LocalDate toDate, BigDecimal rate, BigDecimal amount) {
+        validateVarIsNonNegativeBigDecimal(rate);
+        validateVarIsNonNegativeBigDecimal(amount);
+
+        long noOfDays = calculateNumberOfDays(fromDate, toDate);
+
+        return BigDecimal.valueOf((amount.doubleValue() * noOfDays * rate.doubleValue()) / (365 * 100));
```

Comment:
We should use `BigDecimal` methods for calculation. The whole point of this story is to increase correctness of the calculation result, making calculation on flouting point numbers does not guarantee that.